### PR TITLE
Basic guest-profiler documentation for the book

### PR DESCRIPTION
**Pull request by jameysharp**
This isn't nearly as detailed as the other profiling chapters but I'm struggling to figure out what's important to say. I'd welcome feedback, although of course we can merge this and fill in details later.

This documentation reflects the changes merged in #6362, but that merged after the 9.0 branch was created. Perhaps we should backport that PR to 9.0 so the older CLI flags never appear in a release?

### CODE REVIEW

File: docs/examples-profiling.md [hunk 1]
```diff
@@ -6,8 +6,25 @@ well your Wasm module is performing! From time to time you might want to dive a
...
+operating system. Because Wasmtime is a JIT, that also requires hooks from
+Wasmtime to your platform's native profiling tools.
+
+As a result, Wasmtime support for native profiling is limited to certain
+platforms. See the following sections of this book if you're using these
+platforms:
+
+- On Linux, we support [perf](./examples-profiling-perf.md).
+- For Intel's x86 CPUs on Linux or Windows, we support
+  [VTune](./examples-profiling-vtune.md).
```
Comment (alexcrichton) [inline docs/examples-profiling.md:20]: Mind rewording the seconds around here so the list here can be a 3-bullet list with perf/vtune/guest profiling? I suspect many will scan over this page, see two bullets, and deduce there's two means of profiling Wasmtime. I'm sure many more yet will read further and realize there's a third, but having them all in one organized bulleted list I think makes sense (with all the existing commentary about which is appropriate of course)

**Review by alexcrichton — APPROVED** [review]

File: docs/examples-profiling-guest.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+<https://profiler.firefox.com/>.
+
+To use this profiler with the Wasmtime CLI, pass the
+`--profile=guest[,path[,interval]]` flag.
+
+- `path` is where to write the profile, `wasmtime-guest-profile.json` by default
+- `interval` is the duration between samples, 10ms by default
+
+When used with `--wasm-timeout`, the timeout will be rounded up to the nearest
+multiple of the profiling interval.
```
Comment (fitzgen) [inline docs/examples-profiling-guest.md:14]: This should maybe mention that you can load the profile in perf.firefox.com? Would be nice to include a screenshot as well.

Comment (jameysharp) [reply docs/examples-profiling-guest.md:14]: It definitely should mention that. Also, it does mention it, on line 5. :grin:

A screenshot or two is a good idea but I'm not feeling up to doing that right this moment, so I'd like to merge without it.

File: docs/examples-profiling.md (hunk 1 above)
Comment (jameysharp) [reply docs/examples-profiling.md]: I think it's useful to have the distinction and separation between native and guest profilers, but I've added a bullet point to direct people to keep reading; do you think that takes care of it?

**Review by alexcrichton — APPROVED** [review]
👍

File: docs/examples-profiling-guest.md (hunk 2 above)
Comment (fitzgen) [reply docs/examples-profiling-guest.md:14]: Lol whoops don’t know how I missed that

**Pull request merged**